### ISSUE TITLE
iOS Memory Leak Issue

### ISSUE THREAD

**Issue by NilayDagdemir** [Kotlin/kotlinx.coroutines#2869]
### **New Issue Checklist**
- [x]  I searched for existing GitHub issues

**Info**

I'm using:
```
kotlin: 1.5.20
ktor: 1.6.0
kotlinx-coroutines: 1.4.3-native-mt
```
for a Kotlin Multiplatform project.

### **Describe the bug**
kotlinx-coroutines creates memory leaks in our iOS part of the project. Here are the places of the leaks:

```
>  kotlinx-coroutines-core/common/src/JobSupport.kt:989 (on attachChild(..) function)
>  kotlinx-coroutines-core/common/src/Job.kt:379 (on Job(..) function)
```

And ss for the last line of the memory leaks' stack trace in case you need them:

<img width="697" alt="Kotlinx Coroutines Leaks 1 " src="https://user-images.githubusercontent.com/20637243/129029890-66bd804d-633f-48f5-83c5-910525ed8085.png">
<img width="670" alt="Kotlinx Coroutines Leaks 2" src="https://user-images.githubusercontent.com/20637243/129029893-e847d12a-8193-4218-8125-18c1f0ae42eb.png">

**### Expected behavior**
kotlinx-coroutines shouldn't create any memory leaks in iOS.

**Comment by qwwdfsad** [comment]
It seems like this issue is fixed in 1.5.x-native-mt, please try it and check if the problem still reproduces

**Comment by qwwdfsad** [comment]
`native-mt` is no longer supported, so closing it as won't fix. See https://github.com/Kotlin/kotlinx.coroutines/issues/462#issuecomment-999557659
